pkg/meta/sql.go

```diff
@@ -2194,19 +2247,23 @@ func (m *dbMeta) doLink(ctx Context, inode, parent Ino, name string, attr *Attr)
 		if err = mustInsert(s, &edge{Parent: parent, Name: []byte(name), Inode: inode, Type: n.Type}); err != nil {
 			return err
 		}
-		if updateParent {
-			if _, err := s.Cols("mtime", "ctime", "mtimensec", "ctimensec").Update(&pn, &node{Inode: parent}); err != nil {
-				return err
-			}
-		}
 		if _, err := s.Cols("nlink", "ctime", "ctimensec", "parent").Update(&n, node{Inode: inode}); err != nil {
 			return err
 		}
 		if err == nil {
 			m.parseAttr(&n, attr)
 		}
+		if updateParent {
+			if n, err := s.Cols("mtime", "ctime", "mtimensec", "ctimensec").Update(&pn, &node{Inode: parent}); err != nil || n == 0 {
+				if err == nil {
```

Comment:
we should double check if the parent exists or not for MySQL

Reply:
In rare condition (high concurrency), no column will be real updated. Let me think and find out a way for MySQL.

Reply:
Just read the code, Link operation does not update parent always since it does not change nlink, seems it's ok?
The c/mtimensec update is missing from the original code, fixed.

Reply:
Since we removed the lock, the parent could be delete in the same time as we try to link, so we should make sure that the parent exists when the update affected 0 rows.

Reply:
It does not work as we use repeatable read, even if deleted by other session, you can still get one row in the current transaction.